numba/tests/test_polynomial.py

```diff
@@ -115,3 +131,175 @@ def check(a, **kwargs):
         self.assert_no_domain_change("eigvals", cfunc, (x,))
         # but works fine if type conv to complex first
         cfunc(x.astype(np.complex128))
+
+class TestPolynomial(MemoryLeakMixin, TestCase):
+
+    def test_trimseq_basic(self):
+        pyfunc = trimseq
+        cfunc = njit(trimseq)
+        for i in range(5):
+            coefs = np.array([1] + [0]*i)
+            self.assertPreciseEqual(pyfunc(coefs), cfunc(coefs))
+
+    def test_trimseq_exception(self):
+        cfunc = njit(trimseq)
+
+        self.disable_leak_check()
+
+        with self.assertRaises(TypingError) as raises:
+            cfunc("abc")
+        self.assertIn('The argument "seq" must be array-like',
+                      str(raises.exception))
+
+        with self.assertRaises(TypingError) as e:
+            cfunc(np.arange(10).reshape(5, 2))
+        self.assertIn('Coefficient array is not 1-d',
+                      str(e.exception))
+
+    def test_polyadd_basic(self):
+        pyfunc = polyadd
+        cfunc = njit(polyadd)
+        def inputs():
+            #basic
+            for i in range(5):
+                for j in range(5):
+                    p1 = np.array([0]*i + [1])
+                    p2 = np.array([0]*j + [1])
+                    yield p1, p2
+            #test lists, tuples, scalars
+            yield [1, 2, 3], [1, 2, 3]
+            yield [1, 2, 3], (1, 2, 3)
+            yield (1, 2, 3), [1, 2, 3]
+            yield [1, 2, 3], 3
+            yield 3, (1, 2, 3)
+            #test different dtypes
+            yield np.array([1, 2, 3]), np.array([1.0, 2.0, 3.0])
+            yield np.array([1j, 2j, 3j]), np.array([1.0, 2.0, 3.0])
+            yield np.array([1, 2, 3]), np.array([1j, 2j, 3j])
+
+        for p1, p2 in inputs():
+            self.assertPreciseEqual(pyfunc(p1,p2), cfunc(p1,p2))
+
+    def test_polyadd_exception(self):
```

Comment:
The exception cases are the same for polyadd/polysub/polymul. Can you refactor this code to reduce code duplication? One alternative would be to have a function `_test_poly_exception(self, pyfunc)`:

```python
def _test_polyfunc_exception(self, pyfunc):
    ....

def test_polyadd_exception(self):
    self._test_polyfunc_exception(self, polyadd)

...
```